node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java

```diff
@@ -57,17 +60,19 @@ protected boolean maintain() {
             List<Node> nodesToRemove = retiredNodes.stream().filter(n -> canRemove(n, activeNodes)).collect(Collectors.toList());
             if (nodesToRemove.isEmpty()) continue;
 
+            attempts++;
             try (MaintenanceDeployment deployment = new MaintenanceDeployment(application, deployer, metric, nodeRepository())) {
                 if ( ! deployment.isValid()) continue;
 
                 nodeRepository().nodes().setRemovable(application, nodesToRemove);
                 boolean success = deployment.activate().isPresent();
-                if ( ! success) return success;
+                if ( ! success) continue;
                 String nodeList = nodesToRemove.stream().map(Node::hostname).collect(Collectors.joining(", "));
                 log.info("Redeployed " + application + " to deactivate retired nodes: " +  nodeList);
+                successes++;
             }
         }
-        return true;
+        return attempts == 0 ? 1.0 : ((double)successes / attempts);
```

Comment:
Use `asSuccessFactor` here too?

Reply:
Here is was easier to count successes than failures.